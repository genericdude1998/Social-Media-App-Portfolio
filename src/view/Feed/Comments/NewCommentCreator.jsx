import React from 'react';
import { connect } from 'react-redux';
import { useNavigate } from 'react-router';
import { mapNCCDispatchToProps, mapNCCStateToProps } from '../../mappers/mappers';
import styles from './NCC.css';
import withTheme from '../../theme/withTheme';

const NewCommentCreator = ({onSendNewComment, onSetCommentContent, newCommentError, content, postId, theme}) => {
    const navigate = useNavigate();
    return (
        <div className = {`${styles.baseNCC} ${theme === 'dark' ? styles.dark : null}`}>
            <form className = {styles.baseNCCForm}onSubmit={onSendNewComment(content,navigate, postId)}>
                <label className={`${styles.baseNCCLabel} ${theme === 'dark' ? styles.dark : null}`}>Comment</label>
                <input className={styles.baseNCCInput}type="text" onChange={onSetCommentContent}/>
                <button className={styles.baseNCCButton} type={'submit'}>Send Comment</button>
            </form>
        </div>
    );
}

const ConnectedNCC = connect(mapNCCStateToProps, mapNCCDispatchToProps)(NewCommentCreator)

export default  withTheme(ConnectedNCC);
