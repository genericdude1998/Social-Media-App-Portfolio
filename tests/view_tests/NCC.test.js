import React from "react";
import { shallow } from "enzyme";
import NewCommentCreator from "../../src/view/Feed/Comments/NewCommentCreator";
import {mockContent, mockId} from '../mockValues';

jest.mock('react-router', () => {
    const originalModule = jest.requireActual('react-router');
  
    //Mock the default export and named export 'foo'
    return {
      __esModule: true,
      ...originalModule,
      useNavigate: () => jest.fn(),
    };
  });

let wrapper = shallow(
    <NewCommentCreator.WrappedComponent.WrappedComponent 
        onSendNewComment={jest.fn()}
        onSetCommentContent={jest.fn()}
        content={mockContent}
        postId={mockId}
        theme={'light'}
    />
);

describe('NCC', () => {
    it('should matchSnapshot when light theme', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should matchSnapshot when dark theme', () => {
        wrapper.setProps({theme: 'dark'})
        expect(wrapper).toMatchSnapshot();
    });
});