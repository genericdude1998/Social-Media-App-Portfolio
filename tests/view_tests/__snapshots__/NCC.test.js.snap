// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NCC should matchSnapshot when dark theme 1`] = `
<div
  className="baseNCC dark"
>
  <form
    className="baseNCCForm"
  >
    <label
      className="baseNCCLabel dark"
    >
      Comment
    </label>
    <input
      className="baseNCCInput"
      onChange={[MockFunction]}
      type="text"
    />
    <button
      className="baseNCCButton"
      type="submit"
    >
      Send Comment
    </button>
  </form>
</div>
`;

exports[`NCC should matchSnapshot when light theme 1`] = `
<div
  className="baseNCC null"
>
  <form
    className="baseNCCForm"
  >
    <label
      className="baseNCCLabel null"
    >
      Comment
    </label>
    <input
      className="baseNCCInput"
      onChange={[MockFunction]}
      type="text"
    />
    <button
      className="baseNCCButton"
      type="submit"
    >
      Send Comment
    </button>
  </form>
</div>
`;
